### PR TITLE
Lab 9 qualia4 Maxim Pozdniakov

### PR DESCRIPTION
**Pull request by qualia4**
I also translated all of the examples to Java because I can :/

### CODE REVIEW

**Comment by istepmother** [comment]
Do you think that you are super cool, ganyat` ponty na java??))))
Yes, you are.
Congrats, good job!

File: lecture_notes/09 functions/2-scope.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#include <iostream>
+
+string cities[] = {"Athens", "Roma", "London", "Beijing", "Kiev", "Riga"};
+int length = sizeof(cities)/sizeof(string);
+
+string toUpperCase(string s){
+    for(int i = 0; i < s.length(); i++)
+    {
+        s[i] = toupper(s[i]);
+    }
```
Comment (tshemsedinov) [inline lecture_notes/09 functions/2-scope.cpp:11]: Fix code formatting
```suggestion
string toUpperCase(string s) {
    for (int i = 0; i < s.length(); i++) {
        s[i] = toupper(s[i]);
    }
```

File: lecture_notes/09 functions/2-scope.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    print(cities)
+    print(list(map(f, cities)))
+
+    f = lambda s: s.lower()
+    print(cities)
+    print(list(map(f, cities)))
+
+    f = lambda s: s.upper()
+    cities = ['London', 'Beijing', 'Kiev']
```
Comment (tshemsedinov) [inline lecture_notes/09 functions/2-scope.py:16]: Use nested functions to separate lexical contexts

**Review by tshemsedinov — COMMENTED** [review]
Improve code style and code formattiong consistency

**Comment by tshemsedinov** [comment]
Learn git and use terminal to commit changes

**Pull request closed without merging**